src/features/content-explorer/item-list/ItemListIcon.js

```diff
@@ -1,40 +1,12 @@
-import PropTypes from 'prop-types';
 import React from 'react';
-import { FormattedMessage } from 'react-intl';
-
-import BookmarkIcon from '../../../icons/bookmark-icon';
-import FileIcon from '../../../icons/file-icon';
-import FolderIcon from '../../../icons/folder-icon';
+import PropTypes from 'prop-types';
 
-import ItemTypes from '../item-types';
 import { ItemTypePropType } from '../prop-types';
-import messages from '../messages';
+import IconCell from '../../../elements/common/item/IconCell';
 
-const ItemListIcon = ({ type, extension = '', hasCollaborations = false, isExternallyOwned = false }) => {
-    switch (type) {
-        case ItemTypes.FOLDER: {
-            let titleID = 'personalFolder';
-            if (hasCollaborations) {
-                titleID = 'collaboratedFolder';
-            } else if (isExternallyOwned) {
-                titleID = 'externalFolder';
-            }
-            return (
-                <FolderIcon
-                    isCollab={hasCollaborations}
-                    isExternal={isExternallyOwned}
-                    title={<FormattedMessage {...messages[titleID]} />}
-                />
-            );
-        }
-        case ItemTypes.FILE:
-            return <FileIcon extension={extension} title={<FormattedMessage {...messages.file} />} />;
-        case ItemTypes.BOOKMARK:
-            return <BookmarkIcon title={<FormattedMessage {...messages.bookmark} />} />;
-        default:
-            // Use generic file icon as fallback
-            return <FileIcon title={<FormattedMessage {...messages.file} />} />;
-    }
+const ItemListIcon = ({ type, extension, hasCollaborations = false, isExternallyOwned = false }) => {
+    const rowData = { type, extension, has_collaborations: hasCollaborations, is_externally_owned: isExternallyOwned };
```

Comment:
is there a reason this one is snake case? can we camel case it?

Reply:
I think the snake case comes from an API request

Reply:
oh wait this already returned and transformed from snake case to camelcase. @megansmith-box is right, why do we transform it back to snake?

Reply:
nvm again it looks like it's to satisfy the BoxItem type